src/nfa/operations.cc

```diff
@@ -1298,3 +1293,121 @@ std::optional<mata::Word> Nfa::get_word_from_complement(const Alphabet* alphabet
+            const bool final_excluded_intersects_targets{ nfa_excluded.final.intersects_with(orig_targets_excluded) };
+            subset_set_included_ptr = subset_set_included.insert(std::move(orig_targets_included)).first.operator->();
+            subset_set_excluded_ptr = subset_set_excluded.insert(std::move(orig_targets_excluded)).first.operator->();
+            const auto [target_macrostate_it, macrostate_inserted]{ subset_macrostate_map.emplace(
+                std::make_pair(subset_set_included_ptr, subset_set_excluded_ptr), nfa_lang_difference.num_of_states()
+            ) };
+            subset_macrostate_map_ptr = target_macrostate_it.operator->();
+            const State target_macrostate{ target_macrostate_it->second };
+            nfa_lang_difference.delta.add(macrostate, symbol_advanced_to, target_macrostate);
+            if (macrostate_inserted) {
+                // 'sync_it_excluded_advanced' is true iff there is a transition in the excluded NFA over the symbol
+                //  'symbol_advanced_to'. If sync_it_excluded_advanced == false, the complement of the excluded NFA will
+                //  have a transition over 'symbol_advanced_to' to a "sink state" which is a final state in the
+                //  complement, and therefore must always be final in the language difference.
+                if (final_included_intersects_targets) {
+                    if (subset_set_excluded_ptr == subset_set_excluded_sink_ptr
+                        || (sync_it_excluded_advanced && !final_excluded_intersects_targets)) {
+                        nfa_lang_difference.final.insert(target_macrostate);
+                    }
+                }
+                if (macrostate_discover.has_value()
+                    && !(*macrostate_discover)(
+                        nfa_included, nfa_excluded,
+                        *subset_set_included_ptr, *subset_set_excluded_ptr,
+                        target_macrostate, nfa_lang_difference)) { return nfa_lang_difference; }
+
+                worklist.emplace_back(subset_macrostate_map_ptr);
+            }
+            sync_it_included_advanced = synchronized_iterator_included.advance();
+        }
+    }
+    return nfa_lang_difference;
+}
+
+std::optional<mata::Word> mata::nfa::get_word_from_lang_difference(const Nfa & nfa_included, const Nfa & nfa_excluded) {
+    return lang_difference(nfa_included, nfa_excluded,
+        [&](const Nfa& nfa_included, const Nfa& nfa_excluded,
+            const StateSet& macrostate_included_state_set, const StateSet& macrostate_excluded_state_set,
+            const State macrostate, const Nfa& nfa_lang_difference) {
+            (void)nfa_included, (void)nfa_excluded;
+            (void)macrostate_included_state_set, (void)macrostate_excluded_state_set;
+            (void)macrostate;
+            return nfa_lang_difference.final.empty();
```

Comment:
If I understand this correctly, this forces it to stop when we find some final state, i.e. there is some word, right?